src/main/java/jenkins/plugins/logstash/LogstashBuildWrapper.java

```diff
@@ -48,13 +58,21 @@
 public class LogstashBuildWrapper extends BuildWrapper
 {
 
+  @CheckForNull
```

Comment:
LogstashBuildWrapper is the wrong place. This class is deprecated and only here for backwards compatibility.
You have to put this in the LogstashJobProperty.

Reply:
hmm, thanks, I'll investigate what is this `LogstashJobProperty`. Please check out other comments as I'm more concerned about decisions around script handling.

Reply:
LogstashJobProperty is the place were the script needs to be defined.
LogstashConsoleLogFilter is the place were the LogstashWriter is created when enabled globally, set via JobProperty or for the pipeline step.